apis/opts/memory_reservation.go

```diff
@@ -0,0 +1,15 @@
+package opts
+
+import units "github.com/docker/go-units"
+
+// ParseMemoryReservation parses the memory-reservation param of container.
+func ParseMemoryReservation(memoryReservation string) (int64, error) {
+	if memoryReservation == "" {
+		return 0, nil
+	}
+	result, err := units.RAMInBytes(memoryReservation)
```

Comment:
how about direct return `units.RAMInBytes(memoryReservation)`

Reply:
modified

Reply:
I'm afraid it is not a appropriate idea to return directly.
The upper function should handle the err of RAMInBytes and set memoryReservation to 0.

Reply:
if there is an error, the upper function will always return, will not care about the result.  and will return directly to the run execution failure, so there is no need to set this value.
Is that right?